ndb_client/ndb_client.go

```diff
@@ -44,39 +51,36 @@ func NewNDBClient(username, password, url, caCert string, skipVerify bool) *NDBC
 	return &NDBClient{username, password, url, client}
 }
 
-func (ndbClient *NDBClient) Get(path string) (*http.Response, error) {
-	url := ndbClient.url + "/" + path
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	if err != nil {
-		// fmt.Println(err)
-		return nil, err
+func (ndbClient *NDBClient) NewRequest(method, endpoint string, requestBody interface{}) (*http.Request, error) {
+
+	url := ndbClient.url + "/" + endpoint
+
+	var body io.Reader
+
+	if requestBody != nil {
+		// Serialize the request body to JSON.
+		payload, err := json.Marshal(requestBody)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Make sure this is logged either here (preferably) or at the caller

Reply:
Yes absolutely, it is logged by the common caller for all the API calls (i.e. sendRequest function). Did not log within the ndbClient package because we do not have a reference to the context object here.